3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -0,0 +1,157 @@
+package snowplow
+package enrich
+package common
+package enrichments
+package registry
+
+// Java
+import java.net.URI
+
+// Maven Artifact
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// json4s
+import org.json4s.JValue
+
+// Iglu
+import iglu.client.SchemaKey
+import iglu.client.validation.ProcessingMessageMethods._
+
+// Scala Forex
+import org.joda.money.{Money}
+import org.joda.time.{DateTime, DateTimeZone}
+import com.snowplowanalytics.forex._
+import com.snowplowanalytics.forex.oerclient.{OerClientConfig, DeveloperAccount, AccountType, OerResponseError}
+import com.snowplowanalytics.forex.oerclient.OerResponseError._
+import com.snowplowanalytics.forex.{Forex, ForexConfig}
+import com.snowplowanalytics.forex.ForexLookupWhen._
+//import org.joda.money.BigMoney.convertedTo
+// This project
+import common.utils.ConversionUtils
+import utils.ScalazJson4sUtils
+
+/**
+* Companion object. Lets us create an CurrencyConversionEnrichment
+* instance from a JValue.
+*/
+object CurrencyConversionEnrichmentConfig extends ParseableEnrichment {
+
+  val supportedSchemaKey = SchemaKey("com.snowplowanalytics.snowplow", "currency_conversion_config", "jsonschema", "1-0-0")
+  
+  //Creates an CurrencyConversionEnrichment instance from a JValue
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[CurrencyConversionEnrichment] = {
+    isParseable(config, schemaKey).flatMap( conf => {
+      (for {
+        apiKey        <- ScalazJson4sUtils.extract[String](config, "parameters", "apiKey")
+        baseCurrency  <- ScalazJson4sUtils.extract[String](config, "parameters", "baseCurrency")
+        rateAt        <- ScalazJson4sUtils.extract[String](config, "parameters", "rateAt")
+        enrich        =  CurrencyConversionEnrichment(apiKey, baseCurrency, rateAt)
+      } yield enrich).toValidationNel
+    })
+  }
+}
+
+/**
+* Case class to wrap everything we
+* can extract as converted Transaction
+* Amounts.
+*/
+case class TransactionAmounts(
+  total: String, 
+  tax: String, 
+  shipping: String)
+
+// Object and a case object with the same name
+
+case class CurrencyConversionEnrichment(
+  apiKey: String,
+  baseCurrency: String, 
+  rateAt: String) extends Enrichment {
+
+  val version = new DefaultArtifactVersion("0.1.0")
+  
+  // To convert string value to double
+
+  def toDouble(s: String) = try { 
+    Some(s.toDouble) 
+  } catch { 
+    case _ => None 
+  }
+  
+  // To provide Validation for org.joda.money.Money variable
+  def eitherToValidation(input:Either[OerResponseError, Money]): Validation[String, Double]= {
+    input match {
+      case Right(l) => (l.getAmount().doubleValue()).success[String]
+      case Left(l) => (s"Open Exchange Rates error, message: ${l.errorMessage}").failure
+    }
+  }
+   
+  // To retrieve value from Success/Failure and parse them as string 
+  def tryParse(s: Either[OerResponseError, Money]):String = {
```

Comment:
Remove this function. It is downsampling again. You are taking a Success or Failure boxed value and stripping out the Success or Failure. You **cannot** do this: you cannot take a String that is tagged with Failure and just turn it into a String. We no longer know if we have a success or a failure on our hands.